### PR TITLE
UPDATE: UDP, MATLAB, UI, config, ROI operation

### PR DESCRIPTION
**Pull request by tiborauer**
The number of commits might be misleadingly large; however, some/most of them have been applied to allow the separate submission and re-merge of the SPM stuff. The main changes are the followings:

- UDP now implemented by [pyniexp](https://github.com/tiborauer/pyniexp), which is an integrated solution for both UDP and TCP. The latter will be used for image transfer as already implemented in [pynf/dicomexport](https://github.com/tiborauer/pynf/blob/master/example_dicomexport.py)
- MATLAB processes now (gracefully) terminate after closing the app
- Some UI interfaces/settings have been rearranged and renamed as discussed with @ykoush during our last skype discussion
- The operation to combine ROI signals is now configurable rather than hardcoded 
- ini files have been updated and tested on the [demo data](https://github.com/OpenNFT/OpenNFT_Demo/releases)

### CODE REVIEW

File: mlproc.py [hunk 1]
```diff
@@ -11,56 +11,13 @@
 
 """
 
-import time
+import time, os, signal, sys
```
Comment (espdev) [inline mlproc.py]: It seems os, signal, sys modules are not used in the code

Comment (tiborauer) [reply mlproc.py]: You are right! Sorry! They have been left there from my testing version.

File: mlproc.py [hunk 2]
```diff
@@ -151,10 +108,51 @@ def _start(self):
...
+
+        while self._pid.value != -1:
+            # Is matlab alive?
+            try:
+                eng.version(nargout=0)
+            except me.EngineError:
+                break
+            time.sleep(alive_check_period)
+
+        if eng._check_matlab(): eng.exit()
```
Comment (espdev) [inline mlproc.py]: Is this code works for all supported MATLAB versions and all supported platforms?
In fact, I do not remember why we made that crutch here before, but previous code looked just awfully. :)

Comment (tiborauer) [reply mlproc.py]: I am afraid I cannot test on Mac. However, `_check_matlab()` and `exit()` are standard methods of `MatlabEngine`.

**Comment by da0c** [comment]
We will do testing and a bit adoptation of this PR in seprate branch (tiborauer-master) and then merge it to master.
Thank you a lot!

**Pull request merged**